gammapy/estimators/flux.py

```diff
@@ -195,16 +160,20 @@ def run(self, datasets):
 
         model = self.get_scale_model(models)
 
+        energy_min, energy_max = datasets.energy_ranges
+
         with np.errstate(invalid="ignore", divide="ignore"):
-            result = self.get_reference_flux_values(model.model, energy_min, energy_max)
+            result = self.get_reference_flux_values(
```

Comment:
The general big difficulty for LC of AGN is the change of energy threshold with runs. If 1 dataset is a group of runs within a GTI interval, we will have one threshold per dataset. The question arising is which energy threshold for the whole? If we take the lowest energy threshold (or safe energy threshold), we necessarily do an extrapolation of the model. If we use the highest energy threshold, no extrapolation is done and only the 'measured photons' are used, which is safer at the expense of a strong reduction of the energy range. 
There is no ideal solution and this depends of the Use Case, ie a possible change of the model with time at the extremity of the energy ranges...
I am wondering how to manage this difficulty, as it seems to be 'hardcoded' here. I hope that my comment is clear...

Reply:
The current behaviour is to take the full energy range from the dataset as energy range to quote the flux. However the models is fit taking the `mask_safe` into account. The information on the `mask_safe` energy range is lost later, which is non-ideal. WE have to think about where to keep this information. 

This also means the flux is possibly extrapolated, but I think it's only behaviour that makes sense, because the energy integrated flux must be comparable between the different time intervals, even for varying energy range. 

What is currently not possible is to support a different model per time interval (e.g. varying spectral index...), however this could implemented later on the basis of #3418.

Reply:
@bkhelifi A good option to maintain the energy range info for `FluxPoints` might be to add a `.energy_range_safe` attribute which is extract from the `dataset.energy_range_safe` attribute. When change the data type to a `Map` we could store safe energy ranges depending on position / time etc.

Reply:
This discussion sheds some light also on the question of the `FluxMaps` format for flux points. It can be that the actual safe range is different from one estimate to the other so that technically the energy bins used for fitting could be different e.g. at different times. 
This we can't handle with the `FluxMaps` design but we could do it with a flat table.

One safe solution could be to reject energy bins not fully covered by the `mask`.

Reply:
As I sketched in my comment before, I think one way to handle this could be to return a `Map` in the `energy_range_safe` methods and store the resampled map on the `FluxPoints` as well. This allows us to document how the energy threshold changes with position and time and if needed we could convert the measured fluxes based on the best fit model to the safe energy range afterwards and still store it in a flattened table. Do you see what I mean? I think this approach applies to `FluxMaps` as well...